test/wikidata_ids_decorator_test.rb

```diff
@@ -1,7 +1,62 @@
 require 'test_helper'
 
 describe WikidataIdsDecorator do
-  it 'has a version number' do
-    ::WikidataIdsDecorator::VERSION.wont_be_nil
+  describe 'gem' do
+    it 'has a version number' do
+      ::WikidataIdsDecorator::VERSION.wont_be_nil
+    end
+  end
+end
+
+describe WikidataIdsDecorator::Links do
+  around { |test| VCR.use_cassette(File.basename(url), &test) }
+  let(:response) { Scraped::Request.new(url: url).response }
+  let(:decorator) { WikidataIdsDecorator::Links.new(response: response) }
+
+  describe 'English language links' do
+    let(:url) { 'https://en.wikipedia.org/wiki/Life,_the_Universe_and_Everything' }
+
+    describe 'Wikipedia page with links' do
+      it 'adds a data-wikidata attribute to links' do
+        decorator.body.must_include '<a href="/wiki/Douglas_Adams" title="Douglas Adams" wikidata="Q42">'
+      end
+    end
+
+    describe 'Anchored links' do
+      it 'adds a data-wikidata attribute to anchored links' do
+        decorator.body.must_include '<a href="/wiki/OCLC#Identifiers_and_linked_data" title="OCLC" wikidata="Q190593">'
+      end
+
+      it 'does not add a data-wikidata attribute to anchored links pointing to a place within the same page' do
+        decorator.body.must_include '<a href="#Plot_summary">'
+      end
+    end
+
+    describe 'Redirect links' do
+      it 'it follows the redirect and adds the correct wikidata item id' do
+        decorator.body.must_include '<a href="/wiki/Vogons" class="mw-redirect" title="Vogons" wikidata="Q1076157">'
+      end
+    end
+
+    describe 'Links outside of #bodyContent' do
+      it 'should leave these links alone' do
```

Comment:
This one introduces a mismatch of naming styles. All the other descriptions are asserting what _does_ happen. This one uses the "should" style. We're not quite as proscriptive on this as http://betterspecs.org/#expect, but we should be consistent.